homeassistant/components/speedtestdotnet/__init__.py

```diff
@@ -14,14 +17,58 @@
     DEFAULT_SCAN_INTERVAL,
     DEFAULT_SERVER,
     DOMAIN,
+    SENSOR_TYPES,
     SPEED_TEST_SERVICE,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_SERVER_ID): cv.positive_int,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=timedelta(minutes=DEFAULT_SCAN_INTERVAL)
+                ): vol.All(cv.time_period, cv.positive_timedelta),
+                vol.Optional(CONF_MANUAL, default=False): cv.boolean,
+                vol.Optional(
+                    CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)
+                ): vol.All(cv.ensure_list, [vol.In(list(SENSOR_TYPES))]),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def server_id_valid(server_id):
+    """Check if server_id is valid."""
+    try:
+        api = speedtest.Speedtest()
+        api.get_servers([int(server_id)])
+    except (speedtest.ConfigRetrievalError, speedtest.NoMatchedServers):
+        return False
+
+    return True
+
 
 async def async_setup(hass, config):
-    """Intergation no longer supports importing from config."""
+    """Import integration from config."""
+
+    if DOMAIN in config:
+        # check if server_id is valid
+        if CONF_SERVER_ID in config[DOMAIN] and not await hass.async_add_executor_job(
+            server_id_valid, config[DOMAIN][CONF_SERVER_ID]
+        ):
+            _LOGGER.error("Speedtest server_id is not valid.")
```

Comment:
This test should be done in the config flow. We always want to set up the component. The user should be able to set up another config entry manually.

Reply:
If the test fails we should return `self.async_abort` correct?

Reply:
Yes! :+1: